addons/purchase_stock/models/purchase.py

```diff
@@ -404,5 +404,5 @@ def _find_candidate(self, product_id, product_qty, product_uom, location_id, nam
         args can be merged. If it returns an empty record then a new line will
         be created.
         """
-        lines = self.filtered(lambda l: l.propagate_date == values['propagate_date'] and l.propagate_date_minimum_delta == values['propagate_date_minimum_delta'] and l.propagate_cancel == values['propagate_cancel'])
+        lines = self.filtered(lambda l: l.propagate_date == values['propagate_date'] and l.propagate_date_minimum_delta == values['propagate_date_minimum_delta'] and l.propagate_cancel == values['propagate_cancel'] and l.orderpoint_id == values['orderpoint_id'])
```

Comment:
what about benefit of this task to make something like https://github.com/odoo/odoo/blob/4f226e78930b07dff231e72c8c926273c7111337/addons/stock/models/stock_move.py#L693-L700 ?

Reply:
will conflict too much with arm's branch + if it ain't  broke don't fix it :)